homeassistant/helpers/entity.py

```diff
@@ -199,12 +201,10 @@ def update_ha_state(self, force_refresh=False):
             attr.pop(ATTR_HIDDEN)
 
         # Convert temperature if we detect one
-        if attr.get(ATTR_UNIT_OF_MEASUREMENT) in (TEMP_CELSIUS,
-                                                  TEMP_FAHRENHEIT):
-
-            state, attr[ATTR_UNIT_OF_MEASUREMENT] = \
-                self.hass.config.temperature(
-                    state, attr[ATTR_UNIT_OF_MEASUREMENT])
+        if attr.get(ATTR_UNIT_OF_MEASUREMENT) in (
+                TEMP_CELSIUS, TEMP_FAHRENHEIT) and isinstance(state, Number):
```

Comment:
This is a bug since `state` is supposed to be a string. Best to just try to convert it and ignore it if it fails.

```python
try:
    state = str(self.hass.config.units.temperature(float(state), attr[UNIT_OF_MEASUREMENT]))
except ValueError:
    # Could not convert state to float
    pass
```